hurricane/server/__init__.py

```diff
@@ -153,3 +155,45 @@ def callback_command_exception_check(future: asyncio.Future, webhook_url: str =
             current_loop.stop()
         else:
             logger.info("Execution of management commands was successful. Webhook is not set")
+
+
+def check_databases():
+    for db_name in connections:
+        connection = connections[db_name]
+        cursor = connection.cursor()
+        try:
+            cursor.execute("SELECT (1)")
+            logger.info("Database was checked successfully")
+            return True
+        except Exception as e:
+            logger.warning(f"Database command execution has failed with {e}")
+            return False
+        finally:
+            cursor.close()
+
+
+def count_migrations():
+    number_of_migrations = 0
+    for db_name in connections:
+        connection = connections[db_name]
+        if hasattr(connection, "prepare_database"):
+            connection.prepare_database()
+        executor = MigrationExecutor(connection)
+        targets = executor.loader.graph.leaf_nodes()
+        number_of_migrations += len(executor.migration_plan(targets))
+    return number_of_migrations
+
+
+def check_migrations():
+    while True:
```

Comment:
Does a `while True` block the entire IO loop? That's probably something we don't want to introduce. I'd suggest to have an `event` which starts the actual app server after (optional) management commands (optional) migration probe and run successfully (in that order).